### ISSUE TITLE
False positive `R0903: Too few public methods (0/1)` when using generics and shim imports

### ISSUE THREAD

**Issue by astrojuanlu** [pylint-dev/pylint#8865]
### Bug description

My module does a conditional import on another package for backwards compatibility reasons:

```python
try:
    from package.main import Class_A as ClassA
except ImportError:
    from package.main import ClassA
```

and then creates a derived class:

```python
class ClassB(ClassA[int]):
    """Derived class"""
```

and this produces an error (see below).

I notice that when not attempting to use the generics part everything works fine:

```python
class ClassB(ClassA):
    """Derived class"""
```

Complete source code: https://gist.github.com/astrojuanlu/421e78b189a78504e827aac5f2a4c92e

```
> tree -a -L 1
.
├── .git
├── .gitignore
├── .pylintrc
├── .venv
└── package
> tree 
.
└── package
    ├── __init__.py
    ├── derived.py
    └── main.py

2 directories, 3 files
```

### Configuration

```ini
[DESIGN]
min-public-methods=1
```

### Command used

```shell
pylint package
```

### Pylint output

```shell
************* Module package.derived
package/derived.py:11:0: R0903: Too few public methods (0/1) (too-few-public-methods)

-------------------------------------------------------------------
Your code has been rated at 9.17/10 (previous run: 10.00/10, -0.83)
```

### Expected behavior

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.17/10, +0.83)

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.6
Python 3.10.11 (main, May 23 2023, 16:01:00) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### OS / Environment

macOS Ventura
